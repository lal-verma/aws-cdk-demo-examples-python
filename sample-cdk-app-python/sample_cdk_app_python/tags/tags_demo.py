from aws_cdk import (
    Stack,
    RemovalPolicy,
    aws_lambda as _lambda,
    aws_dynamodb as dynamodb,
    Tags
)

from constructs import Construct

class TagsDemoStack_1(Stack):
    def __init__(self, scope: Construct, construct_id: str) -> None:
        super().__init__(scope, construct_id)
    
        dynamodb.Table(self, 'todo-db', 
                       partition_key=dynamodb.Attribute(name='id', 
                                                        type=dynamodb.AttributeType.STRING),
                       removal_policy=RemovalPolicy.DESTROY,)

        _lambda.Function(self, 'todo-service', 
                        runtime=_lambda.Runtime.PYTHON_3_9,
                        handler='todo_service.handler',
                        code=_lambda.Code.from_asset("../sample-lambda-functions/"),)
           
        


class TagsDemoStack_2(Stack):
    def __init__(self, scope: Construct, construct_id: str) -> None:
        super().__init__(scope, construct_id)
    
        dynamodb.Table(self, 'todo-db', 
                       partition_key=dynamodb.Attribute(name='id', type=dynamodb.AttributeType.STRING),
                       removal_policy=RemovalPolicy.DESTROY,)

        todo_service_function = _lambda.Function(self, 'todo-service', 
                        runtime=_lambda.Runtime.PYTHON_3_9,
                        handler='todo_service.handler',
                        code=_lambda.Code.from_asset("../sample-lambda-functions/"),)
           
        Tags.of(todo_service_function).add('microservice', 'todo-service-v2')
