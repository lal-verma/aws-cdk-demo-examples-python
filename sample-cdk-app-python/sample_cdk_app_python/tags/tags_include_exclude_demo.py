from aws_cdk import (
    Stack,
    RemovalPolicy,
    aws_lambda as _lambda,
    aws_dynamodb as dynamodb,
    Tags
)

from constructs import Construct

class TagsExcludeDemoStack(Stack):
    def __init__(self, scope: Construct, construct_id: str) -> None:
        super().__init__(scope, construct_id)
    

        todoservice_function = _lambda.Function(self, 'todo-service', 
                        runtime=_lambda.Runtime.PYTHON_3_9,
                        handler='todo_service.handler',
                        code=_lambda.Code.from_asset("../sample-lambda-functions/"),)
           

        Tags.of(todoservice_function).add("microservice", "todo-service", 
                          exclude_resource_types=["AWS::IAM::Role"]);    


class TagsIncludeDemoStack(Stack):
    def __init__(self, scope: Construct, construct_id: str) -> None:
        super().__init__(scope, construct_id)
    

        todoservice_function = _lambda.Function(self, 'todo-service', 
                        runtime=_lambda.Runtime.PYTHON_3_9,
                        handler='todo_service.handler',
                        code=_lambda.Code.from_asset("../sample-lambda-functions/"),)
           

        Tags.of(todoservice_function).add("microservice","todo-service", 
                          include_resource_types=["AWS::Lambda::Function"]);    