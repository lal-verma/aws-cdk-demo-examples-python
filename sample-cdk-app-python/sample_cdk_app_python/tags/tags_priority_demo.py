from aws_cdk import (
    Stack,
    RemovalPolicy,
    aws_lambda as _lambda,
    aws_dynamodb as dynamodb,
    Tags
)

from constructs import Construct

class TagsPriorityDemoStack(Stack):
    def __init__(self, scope: Construct, construct_id: str) -> None:
        super().__init__(scope, construct_id)
    
        dynamodb.Table(self, 'todo-db', 
                       partition_key=dynamodb.Attribute(name='id', type=dynamodb.AttributeType.STRING),
                       removal_policy=RemovalPolicy.DESTROY,)

        todo_service_function = _lambda.Function(self, 'todo-service', 
                        runtime=_lambda.Runtime.PYTHON_3_9,
                        handler='todo_service.handler',
                        code=_lambda.Code.from_asset("../sample-lambda-functions/"),)
           
        Tags.of(self).add('microservice', 'todo-service')

        Tags.of(todo_service_function).add('microservice', 'todo-service-v2', priority=90);    