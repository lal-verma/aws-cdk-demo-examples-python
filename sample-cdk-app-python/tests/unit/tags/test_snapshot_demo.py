import aws_cdk as core
import aws_cdk.assertions as assertions

from sample_cdk_app_python.tags.tags_demo import (TagsDemoStack_1)


def test_function_snapshot(snapshot):
    app = core.App()
    stack = TagsDemoStack_1(app, "tags-demo")
    template = assertions.Template.from_stack(stack)

    assert template.to_json() == snapshot
                                    