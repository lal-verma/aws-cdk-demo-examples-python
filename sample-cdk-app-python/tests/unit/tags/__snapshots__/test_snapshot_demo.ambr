# name: test_function_snapshot
  dict({
    'Parameters': dict({
      'BootstrapVersion': dict({
        'Default': '/cdk-bootstrap/hnb659fds/version',
        'Description': 'Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]',
        'Type': 'AWS::SSM::Parameter::Value<String>',
      }),
    }),
    'Resources': dict({
      'tododbA0F8A3AC': dict({
        'DeletionPolicy': 'Delete',
        'Properties': dict({
          'AttributeDefinitions': list([
            dict({
              'AttributeName': 'id',
              'AttributeType': 'S',
            }),
          ]),
          'KeySchema': list([
            dict({
              'AttributeName': 'id',
              'KeyType': 'HASH',
            }),
          ]),
          'ProvisionedThroughput': dict({
            'ReadCapacityUnits': 5,
            'WriteCapacityUnits': 5,
          }),
          'Tags': list([
            dict({
              'Key': 'microservice',
              'Value': 'todo-service',
            }),
          ]),
        }),
        'Type': 'AWS::DynamoDB::Table',
        'UpdateReplacePolicy': 'Delete',
      }),
      'todoservice099F38C7': dict({
        'DependsOn': list([
          'todoserviceServiceRole296BAF91',
        ]),
        'Properties': dict({
          'Code': dict({
            'S3Bucket': dict({
              'Fn::Sub': 'cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}',
            }),
            'S3Key': '4ae2188b6aa47f4a781720f62970a579a00d1e9856aef247014d2488e14d790e.zip',
          }),
          'Handler': 'todo_service.handler',
          'Role': dict({
            'Fn::GetAtt': list([
              'todoserviceServiceRole296BAF91',
              'Arn',
            ]),
          }),
          'Runtime': 'python3.9',
          'Tags': list([
            dict({
              'Key': 'microservice',
              'Value': 'todo-service',
            }),
          ]),
        }),
        'Type': 'AWS::Lambda::Function',
      }),
      'todoserviceServiceRole296BAF91': dict({
        'Properties': dict({
          'AssumeRolePolicyDocument': dict({
            'Statement': list([
              dict({
                'Action': 'sts:AssumeRole',
                'Effect': 'Allow',
                'Principal': dict({
                  'Service': 'lambda.amazonaws.com',
                }),
              }),
            ]),
            'Version': '2012-10-17',
          }),
          'ManagedPolicyArns': list([
            dict({
              'Fn::Join': list([
                '',
                list([
                  'arn:',
                  dict({
                    'Ref': 'AWS::Partition',
                  }),
                  ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
                ]),
              ]),
            }),
          ]),
          'Tags': list([
            dict({
              'Key': 'microservice',
              'Value': 'todo-service',
            }),
          ]),
        }),
        'Type': 'AWS::IAM::Role',
      }),
    }),
    'Rules': dict({
      'CheckBootstrapVersion': dict({
        'Assertions': list([
          dict({
            'Assert': dict({
              'Fn::Not': list([
                dict({
                  'Fn::Contains': list([
                    list([
                      '1',
                      '2',
                      '3',
                      '4',
                      '5',
                    ]),
                    dict({
                      'Ref': 'BootstrapVersion',
                    }),
                  ]),
                }),
              ]),
            }),
            'AssertDescription': "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          }),
        ]),
      }),
    }),
  })
# ---
